src/main/java/org/hisp/dhis/tasksets/tracker/TrackerCapture_getEntitiesOrderedByEnrolledAt.java

```diff
@@ -0,0 +1,116 @@
+package org.hisp.dhis.tasksets.tracker;
+
+import org.hisp.dhis.cache.Program;
+import org.hisp.dhis.cache.TrackedEntityAttribute;
+import org.hisp.dhis.cache.User;
+import org.hisp.dhis.common.ValueType;
+import org.hisp.dhis.response.dto.ApiResponse;
+import org.hisp.dhis.tasks.tracker.tei.GetEntitiesTask;
+import org.hisp.dhis.tasks.tracker.tei.oldapi.GetTeiTask;
+import org.hisp.dhis.tasksets.DhisAbstractTaskSet;
+import org.hisp.dhis.utils.Randomizer;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+/**
+ * @author Marc Pratllusa <marc@dhis2.org>
+ */
+public class TrackerCapture_getEntitiesOrderedByEnrolledAt
+    extends DhisAbstractTaskSet
+{
+    private static final String NAME = "TrackerCapture: get entities ordered by enrolment date";
+
+    HashMap<String, List<TrackedEntityAttribute>> attributes = new HashMap<>();
+
+    public TrackerCapture_getEntitiesOrderedByEnrolledAt(int weight )
+    {
+        super( NAME, weight );
+    }
+
+    @Override
+    public String getName()
+    {
+        return NAME;
+    }
+
+    @Override
+    public String getType()
+    {
+        return "http";
+    }
+
+    @Override
+    public void execute()
+        throws Exception
+    {
+        Randomizer rnd = getNextRandomizer( getName() );
+        Program program = rnd.randomElementFromList( this.entitiesCache.getTrackerPrograms() );
+        User user = getUser(rnd);
+        String ou = rnd.randomElementFromList( user.getOrganisationUnits() );
+
+        String url = String
+                .format( "?orgUnit=%s&program=%s&order=enrolledAt", ou, program.getId());
+        ApiResponse response = new GetEntitiesTask( 1, url, user.getUserCredentials(), "get entities, order by enrolment date", rnd )
+            .executeAndGetResponse();
+
+        List<ArrayList> rows = response.extractList( "rows" );
+
+        if ( rows != null && !rows.isEmpty() )
+        {
+            ArrayList row =rnd.randomElementFromList( rows );
+
+            String teiId = row.get( 0 ).toString();
+
+            new GetTeiTask( teiId, user.getUserCredentials(), rnd ).execute();
+        }
+
```

Comment:
I am not sure I understand why this endpoint call is here.
I know we are doing something similar in other taskset, but if this is not something that we want to be there for a purpose, I would suggest to remove it in order to have the simplest and clearest test as possible

Reply:
Now I tried to simplified as much as possible ;)